docs/topics/api-guidelines-build-for-multiplatform.md

```diff
@@ -0,0 +1,84 @@
+
+Place APIs in the broadest relevant source set, using the following priority:
+
+* The `commonMain` source set
+* Intermediate source sets
+* Platform-specific source sets
+
+> To learn more about the source sets of Kotlin Multiplatform projects, see [Hierarchical project structure](multiplatform-hierarchy.md).
+>
+{style="tip"}
+
+APIs in the `commonMain` source set are available to all platforms the library supports.
+Aim to place most of your library’s API in the `commonMain` source set.
+
+If some platforms don’t support certain APIs, use [intermediate source sets](multiplatform-discover-project.md#intermediate-source-sets) to target specific platforms.
+For example, you can create a `concurrent` source set for targets that support multi-threading, or a `nonJvm` source set for all non-JVM targets.
+
+Finally, for platform-specific APIs, use source sets like `androidMain`.
+
+## Ensure consistent behavior across platforms
+
+To ensure your library behaves consistently on all supported platforms,
+APIs in a multiplatform library should accept the same range of valid inputs, perform the same actions,
+and return the same results across all platforms.
+Similarly, the library should treat invalid inputs uniformly, and report errors or throw exceptions consistently on all platforms.
+
+Inconsistent behavior makes the library difficult to use, and forces users to add conditional logic in common code to manage platform-specific differences.
+
+You can use [expect-actual declarations](multiplatform-expect-actual.md) to declare functions in common code with
+platform-specific implementations with full access to the native APIs of each platform.
+These implementations must also have the same behavior to ensure that they can be used reliably from common code.
+
+When APIs behave consistently across platforms, they only need to be documented once, in the `commonMain` source set.
+
+> If platform differences are unavoidable, such as when one platform
+> supports a broader set of inputs, minimize them where possible. For example, you might not want to limit one platform’s functionality to match the others. In such cases, clearly document the specific differences.
+>
+> {style=”note”}
+
+## Test on all platforms
+
+Multiplatform libraries can have [multiplatform tests](https://www.jetbrains.com/help/kotlin-multiplatform-dev/multiplatform-run-tests.html) written in common code, which can run on all platforms.
+Executing this common test suite regularly on your supported platforms can ensure that the library behaves correctly and consistently.
+
+Regularly testing Kotlin/Native targets across all published platforms can be challenging.
+However, to ensure broader compatibility, consider publishing the library for all targets it can support [using a tiered approach](native-target-support.md#for-library-authors) for testing.
```

Comment:
```suggestion
However, to ensure broader compatibility, consider publishing the library for all targets it can support [using a tiered approach](native-target-support.md#for-library-authors) to testing.
```

Reply:
hmm I think "for" works better here - "it is used for testing"
Alternatively we could say "to test" but then it would become less concise. 

However, to ensure broader compatibility, consider publishing the library for all targets it can support [using a tiered approach](https://github.com/Kotlin/api-guidelines/pull/native-target-support.md#for-library-authors) to test compatibility.